tiledb/sm/query/sparse_index_reader_base.cc

```diff
@@ -336,16 +372,16 @@ Status SparseIndexReaderBase::read_and_unfilter_coords(
     }
   }
 
-  if (include_timestamps) {
+  if (include_timestamps || !partial_overlap_condition_.empty()) {
     std::vector<std::string> timestamps_names = {constants::timestamps};
     RETURN_CANCEL_OR_ERROR(
         read_attribute_tiles(timestamps_names, result_tiles, true));
     RETURN_CANCEL_OR_ERROR(
         unfilter_tiles(constants::timestamps, result_tiles, true));
   }
 
-  if (!condition_.empty()) {
-    // Read and unfilter tiles for querty condition.
+  if (!condition_.empty() || !partial_overlap_condition_.empty()) {
```

Comment:
!partial_overlap_condition_.empty() can be removed... Timestamps should be loaded above.